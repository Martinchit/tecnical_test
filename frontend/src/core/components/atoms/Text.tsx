import styled from 'styled-components';

export type WeightProps = 'light' | 'normal' | 'bold' | 'extra-bold';

export interface TextProps {
  fontSize?: number;
  fontWeight?: WeightProps;
  fontColor?: string;
}

const getFontSize = (props) => () => {
  const { fontSize } = props;
  return fontSize ? `${fontSize}em` : '1em';
};

const getFontWeight = (props) => () => {
  const { fontWeight } = props;
  switch (fontWeight) {
    case 'light':
      return '200';
    case 'normal':
      return '400';
    case 'bold':
      return '600';
    case 'extra-bold':
      return '800';
    default:
      return '400';
  }
};

const getFontColor = (props) => () => {
  const { fontColor } = props;
  return fontColor || '#000';
};

const setDefaultValue = () => 'margin: 0px;padding: 0px;border:0';

export const Text = styled.p<TextProps>`
  font-size: ${getFontSize};
  font-weight: ${getFontWeight};
  color: ${getFontColor};
  ${setDefaultValue}
`;
