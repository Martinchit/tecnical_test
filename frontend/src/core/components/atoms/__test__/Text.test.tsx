import React from 'react';
import { shallow } from 'enzyme';
import 'jest-styled-components';

import { Text } from '../Text';

describe('Text', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Text />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders font-weight 200', () => {
    const wrapper = shallow(<Text fontWeight="light" />);
    expect(wrapper).toHaveStyleRule('font-weight', '200');
  });

  it('renders font-weight 400', () => {
    const wrapper = shallow(<Text fontWeight="normal" />);
    expect(wrapper).toHaveStyleRule('font-weight', '400');
  });

  it('renders font-weight 600', () => {
    const wrapper = shallow(<Text fontWeight="bold" />);
    expect(wrapper).toHaveStyleRule('font-weight', '600');
  });

  it('renders font-weight 800', () => {
    const wrapper = shallow(<Text fontWeight="extra-bold" />);
    expect(wrapper).toHaveStyleRule('font-weight', '800');
  });

  it('renders current font-size', () => {
    const wrapper = shallow(<Text fontSize={1} />);
    expect(wrapper).toHaveStyleRule('font-size', '1em');
  });
});
