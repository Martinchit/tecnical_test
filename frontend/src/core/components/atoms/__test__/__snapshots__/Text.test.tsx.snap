// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text renders correctly 1`] = `
.c0 {
  font-size: 1em;
  font-weight: 400;
  color: #000;
  margin: 0px;
  padding: 0px;
  border: 0;
}

<p
  className="c0"
/>
`;
